modules/build/src/test/scala/scala/build/tests/NativeBuilderHelperTests.scala

```diff
@@ -43,117 +42,117 @@ class NativeBuilderHelperTests extends munit.FunSuite {
     inputs.withBuild(defaultOptions, buildThreads, bloopConfig) { (root, _, maybeBuild) =>
       val build = maybeBuild.toOption.get.successfulOpt.get
 
-      val nativeConfig  = build.options.scalaNativeOptions.config
+      val config        = build.options.scalaNativeOptions.configCliOptions
       val nativeWorkDir = build.options.scalaNativeOptions.nativeWorkDir(root, "native-test")
       val destPath      = nativeWorkDir / s"main${if (Properties.isWin) ".exe" else ""}"
       // generate dummy output
       os.write(destPath, Random.alphanumeric.take(10).mkString(""), createFolders = true)
 
-      val changed =
-        NativeBuilderHelper.shouldBuildIfChanged(build, nativeConfig, destPath, nativeWorkDir)
-      expect(changed)
+      val cacheData =
+        NativeBuilderHelper.getCacheData(build, config, destPath, nativeWorkDir)
+      expect(cacheData.changed)
     }
   }
 
   test("should not rebuild the second time") {
     inputs.withBuild(defaultOptions, buildThreads, bloopConfig) { (root, _, maybeBuild) =>
       val build = maybeBuild.toOption.get.successfulOpt.get
 
-      val nativeConfig  = build.options.scalaNativeOptions.config
+      val config        = build.options.scalaNativeOptions.configCliOptions
```

Comment:
NIT (same for the others below)
```suggestion
      val config        = build.options.scalaNativeOptions.configCliOptions()
```